test-utils/README.md

```diff
@@ -0,0 +1,42 @@
+# `libcrux-test-utils`
+
+This crate contains tools for testing, tracing, profiling and benchmarking.
+
+## Tracing (`tracing`)
+
+This module provides tools for annotating functions such that calls are traced.
+That means that whenever the function is called and when it returns, an entry
+is written to the specified trace.
+
+Usually, that trace will be a static variable with interior mutability. If in
+doubt, define it something like this:
+
+```rust
+lazy_static::lazy_static! {
+    static ref TRACE: MutexTrace<&'static str, Instant> = Default::default();
+}
+```
+
+The `MutexTrace` can be defined and used as a static variable without unsafe, but,
+depending on the setting, my introduce too much overhead. Any type that implements
+`Trace` works.
+
+Then, annotate a function like this:
+
+```rust
+#[libcrux_macros::trace_span("trace_me", TRACE)]
+fn this_function_is_traced() {
+  // ...
+}
+```
+
+The macro is called `trace_span` because it traces a start and end. There also
+are on-the-fly tracing facilities.
+
+After the code in question ran, the trace can be inspected. Due to the use
+of  interior mutability, there is no generic way to get a reference to the
```

Comment:
That's a little unfortunate. But how is related to interior mutability? In the usages here it seems like this is because we have a static refence that we can't consume/drop?

Reply:
Ah, yes, there are two things here.

I think for basically all trace inspection tasks it would be enough to have references to the trace entries. However, the interior mutability types mean that we can't get references into the data, so we have to use owned values. And yes, the only way to get an owned value from a static reference is to clone it.

Actually I suppose we could also use a `Mutex<Option<T>>` and then `.take()` that, or create a new `T` and use `mem::swap`, but I'm not sure that would be a lot better. This is only for testing, after all the benchmarks ran, so I am not sure we need to be super performant here.

Reply:
Yeah it's not a real issue, more an annoyance that it seems necessary.